### ISSUE TITLE
Unquoted Boolean Attribute in Self-closing input tag causes render to reject

### ISSUE THREAD

**Issue by bennypowers** [Polymer/lit-html#870]
# Description
Using a boolean attribute in a self-closing input tag causes render to reject.

# Steps to Reproduce

1. Write this code

```ts
import {LitElement, html} from 'lit-element';

class MyElement extends LitElement {
  render() {
    // return html`<input type="checkbox" ?checked="${true}"/>`
    return html`<input type="checkbox" ?checked=${true}/>`
    // return html`<input type="checkbox" ?checked="${true}"></input>`
    // return html`<input type="chec kbox" ?checked=${true}></input>`
  }
}

customElements.define('my-element', MyElement);
```

2. See this output...

## Live Reproduction Link

https://stackblitz.com/edit/lit-element-for-you-4mwwnj?file=my-element.ts

## Expected Results
I would expect the input to render

## Actual Results
render promise rejects `Error: Boolean attributes can only contain a single expression`

## Browsers Affected

- [x] Chrome
- [x] Firefox
- [ ] Edge
- [ ] Safari 11
- [ ] Safari 10
- [ ] IE 11

**Comment by justinfagnani** [comment]
Move to lit-html, since this should be local to it.

**Comment by blikblum** [comment]
Seems the same underlying issue when passing an object to a property in [a self closing custom element](https://github.com/Polymer/lit-html/issues/630#issuecomment-452542209)

**Comment by jridgewell** [comment]
A solution I can see is to add quotes to unquoted attributes in the `TemplateResult.p.getHTML`, but this would cause `<div attr=first${'middle'}last>` to change behavior (before, you'd have `'first' + 'middle' + 'last'`, now `'first' + 'middle'` and a `last` attribute).

**Comment by justinfagnani** [comment]
This is fixed in 2.0
